### PR TITLE
Support bench duration && warmup

### PR DESCRIPTION
**Pull request by YangKian**

### CODE REVIEW

**Review by s12f — APPROVED** [review]

File: app/src/main/java/io/hstream/tools/WriteReadBench.java [hunk 1]
```diff
@@ -75,7 +75,7 @@ public static void main(String[] args) throws Exception {
 
     lastReportTs = System.currentTimeMillis();
     long terminateTs =
-        options.benchmarkDuration == Long.MAX_VALUE
+        Long.MAX_VALUE - lastReportTs < options.benchmarkDuration * 1000L
```
Comment (s12f) [inline app/src/main/java/io/hstream/tools/WriteReadBench.java:78]: Would ``options.benchmarkDuration * 1000L`` not overflow if ``options.benchmarkDuration`` is the default value(Long.MAX_VALUE)?

Comment (YangKian) [reply app/src/main/java/io/hstream/tools/WriteReadBench.java:78]: Oh, you are right, that's a problem. Do you have any suggestions for this situation? I can't find an elegant way to deal with this problem. 

Changing the units of `benchmarkDuration` to `ms` is always right, but it is not user friendly :thinking: 
Or, convert the default value to `Integer.MAX_VALUE` as before? I think it is long enough for a test case to express never stop.

Comment (s12f) [reply app/src/main/java/io/hstream/tools/WriteReadBench.java:78]: @YangKian I think you can add `benchmarkDuration` <= 0 condition if user want to disable Duration strategy, e.g.:
```
     // a super big options.benchmarkDuration is realy pointless
     long terminateTs =  options.benchmarkDuration > 0 
             ? lastReportTs + options.benchmarkDuration * 1000L 
             : 0;
     
     ......

      if (terminateTs > 0 && now >= terminateTs) {
        terminateFlag.set(true);
        break;
      }
```
and the default value is 0, which could solve your problem.

Comment (YangKian) [reply app/src/main/java/io/hstream/tools/WriteReadBench.java:78]: The question is, in this expression `lastReportTs + options.benchmarkDuration * 1000L`, even if `benchmarkDuration * 1000` doesn't overflow, the total expression may overflow :sob:

Comment (YangKian) [reply app/src/main/java/io/hstream/tools/WriteReadBench.java]: I found another way to do this, please see the new update.@s12f

**Review by s12f — APPROVED** [review]

**Pull request merged**